### PR TITLE
IGAPP-1145: Locally deploy rasa widget

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web/src/components/ChatBotWidget.tsx [hunk 1]
```diff
@@ -18,7 +19,7 @@ const ChatBotWidget = (): ReactElement => {
   useEffect(() => {
     const script = document.createElement('script')
 
-    script.src = 'https://unpkg.com/@rasahq/rasa-chat'
+    script.src = `/rasa-widget/chatbot.js`
```
Comment (SteffiStuffel) [inline web/src/components/ChatBotWidget.tsx:22]: Widget is released unter Apache 2.0 licence, so copying it to our MIT licenced project should be fine.

File: web/www/rasa-widget/chatbot.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/*! For license information please see widget.js.LICENSE.txt */
```
Comment (SteffiStuffel) [inline web/www/rasa-widget/chatbot.js:1]: It is just the same file as from unpkg, but changed the font, as it was a google font.

**Review by f1sh1918 — COMMENTED** [review]
Implementation looks fine, but i get an invalid certificate error

image

**Review by LeandraH — APPROVED** [review]
Looks good, tested in Firefox.

File: web/src/components/ChatBotWidget.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const ChatBotWidget = (): ReactElement => {
...
```
Comment (steffenkleinle) [inline web/src/components/ChatBotWidget.tsx:21]:
```suggestion
    script.src = '/rasa-widget/chatbot.js'
```

File: web/www/rasa-widget/chatbot.js (hunk 2 above)
Comment (steffenkleinle) [reply web/www/rasa-widget/chatbot.js:1]: Does anyone whats a good practice to keep scripts like this updated? Should we just have a reminder in like 6 months to check if something changed?

**Review by steffenkleinle — APPROVED** [review]
Tested on firefox, looks good

File: web/www/rasa-widget/chatbot.js (hunk 2 above)
Comment (LeandraH) [reply web/www/rasa-widget/chatbot.js:1]: Good question, I don't know one and a quick Google doesn't turn up an obvious answer either. A reminder every six months to check all copied scripts (maybe with a growing check list of files to check?) seems like a decent solution for now.

**Comment by LeandraH** [comment]
We should check if the errors @f1sh1918 got are fixed after the Salt PR is merged.

**Comment by SteffiStuffel** [comment]
> Implementation looks fine, but i get an invalid certificate error
> 

Cannot reproduce. I do not get this error. Also this seems to be an error of the certificate twb created, so we cannot change this anyways. But it seems they have fixed this already.

**Pull request merged**